assets/onboarding/use-fullscreen.jsx

```diff
@@ -1,16 +1,27 @@
-import { useEffect } from '@wordpress/element';
+import { useLayoutEffect } from '@wordpress/element';
+
+/**
+ * Apply fullscreen view by hiding wp-admin elements via CSS.
+ *
+ * Allows setting additional classes on the body element.
+ * Fullscreen and classes are added when the component is mounted, and removed when unmounted.
+ *
+ * @param {string[]} bodyClasses Additional classes to be set.
+ */
+export function useFullScreen( bodyClasses = [] ) {
+	bodyClasses.push( 'sensei-wp-admin-fullscreen' );
 
-export function useFullScreen() {
 	function setupGlobalStyles() {
 		toggleGlobalStyles( true );
 		return toggleGlobalStyles.bind( null, false );
 	}
 
 	function toggleGlobalStyles( enabled ) {
-		document.body.classList.toggle( 'sensei-wp-admin-fullscreen', enabled );
-		document.body.classList.toggle( 'sensei-color', enabled );
+		if ( enabled ) document.body.classList.add( ...bodyClasses );
+		else document.body.classList.remove( ...bodyClasses );
+
 		document.documentElement.classList.toggle( 'wp-toolbar', ! enabled );
 	}
 
-	useEffect( setupGlobalStyles, [] );
+	useLayoutEffect( setupGlobalStyles, [] );
```

Comment:
Out of curiosity, did you test whether this solves the blink issue? Maybe it could be the reason. So we could remove that PHP workaround :)

Reply:
It solves a different blink issue :) `html.wp-toolbar` adds some top margin to the page, and I couldn't find a filter to remove that toolbar class wp-admin, so useFullscreen does this. But useEffect would run after the component is rendered, so there was a (much less noticeable) jump of the header upon load.